### ISSUE TITLE
function body uses variable which out of scope 

### ISSUE THREAD

**Issue by WindSoilder** [nushell/nushell#9746]
### Describe the bug

Continue from comment: https://github.com/nushell/nushell/issues/9724#issuecomment-1643448200

### How to reproduce

Run the following code in repl or script
```nushell
let b = 3
def c [] { echo $b }
c
```
It outputs `3`

### Expected behavior

I expected nu to returns error which says that `$b` is not found

### Configuration

| key                | value                                                   |
| ------------------ | ------------------------------------------------------- |
| version            | 0.82.1                                                  |
| branch             | nightly                                                 |
| commit_hash        | bd0032898fe4697f8f1f9e48b3bb904f87608fcb                |
| build_os           | linux-x86_64                                            |
| build_target       | x86_64-unknown-linux-gnu                                |
| rust_version       | rustc 1.68.2 (9eb3afe9e 2023-03-27)                     |
| rust_channel       | 1.68.2-x86_64-unknown-linux-gnu                         |
| cargo_version      | cargo 1.68.2 (6feb7c9cf 2023-03-26)                     |
| build_time         | 2023-07-12 02:15:14 +00:00                              |
| build_rust_channel | release                                                 |
| allocator          | mimalloc                                                |
| features           | default, sqlite, static-link-openssl, trash, which, zip |
| installed_plugins  |                                                         |

### Additional context

The following works as expected, although the error message is not fine.
```nushell
 do { 
    let b = 3
    def c [] { echo $b }
    c 
 }
```

**Comment by jntrnr** [comment]
After discussing with folks during the Nushell meeting, it looks like the system does allow `def` to close over variables. Your examples should work.

I've fixed the `do` case in #9764

**Comment by WindSoilder** [comment]
Ok, so I'd closing this
